src/java.base/share/classes/java/security/MessageDigestSpi.java

```diff
@@ -205,7 +205,11 @@ protected int engineDigest(byte[] buf, int offset, int len)
      */
     public Object clone() throws CloneNotSupportedException {
         if (this instanceof Cloneable) {
-            return super.clone();
+            var o = super.clone();
```

Comment:
You could instead cast the return value of `super.clone` to `MessageDigestSpi` and avoid the casts below.

Reply:
I like that suggestion.